cmd/run.go

```diff
@@ -55,23 +56,25 @@ func Run() (err error) {
 		for _, pluginPkg := range plugins {
 			if pluginPkg.Name == servicePluginName {
 				if !pluginPkg.Available {
-					logger.Error("Requested plugin that is not installed: " + pluginPkg.Name)
-					continue
+					logger.Error(fmt.Sprintf("Requested plugin that is not installed: " + pluginPkg.Name))
+					return 4
 				}
 				client := newClient(pluginPkg.Command)
 				defer closeClient(pluginPkg, client)
 
 				// Connect via RPC
 				var rpcClient hcplugin.ClientProtocol
-				rpcClient, err = client.Client()
+				rpcClient, err := client.Client()
 				if err != nil {
-					return err
+					logger.Error(fmt.Sprintf("internal error while initializing RPC client: %s", err))
+					return 3
 				}
 				// Request the plugin
 				var rawPlugin interface{}
 				rawPlugin, err = rpcClient.Dispense(shared.PluginName)
 				if err != nil {
-					logger.Error(err.Error())
+					logger.Error(fmt.Sprintf("internal error while dispensing RPC client: %s", err.Error()))
+					return 3
```

Comment:
do we want these numbers to be constants at the top?  We can then trace usage throughout the file and other places.  With single entries we'd have to look for integers and that can be interesting. wdyt?

Reply:
sure, good idea

Reply:
@jmeridth applied